plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotMenuActions.java

```diff
@@ -0,0 +1,174 @@
+import org.eclipse.swtbot.swt.finder.SWTBot;
+import org.eclipse.swtbot.swt.finder.exceptions.WidgetNotFoundException;
+import org.eclipse.swtbot.swt.finder.waits.DefaultCondition;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotShell;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTree;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.hamcrest.Matcher;
+
+/**
+ * SWTBot utility methods that perform general file actions.
+ */
+public final class SwtBotMenuActions {
+
+  public static String getEditorText(final SWTWorkbenchBot bot, String title) {
+    SWTBotEclipseEditor textEditor = bot.editorByTitle(title).toTextEditor();
+    return textEditor.getText();
+  }
+
+  /**
+   * Opens the Debug Configurations... perspective.
+   */
+  public static void openDebugConfiguration(final SWTBot bot) {
+    SwtBotTestingUtilities.performAndWaitForWindowChange(bot, new Runnable() {
+      @Override
+      public void run() {
+        // Be sure the main window has focus
+        setFocusMainShell(bot);
+
+        bot.menu("Run").menu("Debug Configurations...").click();
+      }
+    });
+  }
+
+  public static void openDebugPerspective(final SWTBot bot) {
+    // Set timeout low, in case the shell is not a workbench window
+    SwtBotTimeoutManager.setTimeout(500);
+    try {
+      bot.menu("Window").menu("Open Perspective").menu("Debug").click();
+    } catch (WidgetNotFoundException e) {
+      // Try "Other" in case we are in a perspective that doesn't list Debug
+      // as a perspective shortcut, such as the Resource perspective.
+      bot.menu("Window").menu("Open Perspective").menu("Other...");
+      bot.list().select("Debug");
+      bot.button("OK").click();
+    }
+    // Restore original timeout
+    SwtBotTimeoutManager.setTimeout();
+  }
+
+  /**
+   * Open debug perspective and terminate the process and wait for it be terminated.
+   */
+  public static void openDebugPerspectiveAndTerminateProcess(SWTWorkbenchBot bot) {
+    openDebugPerspective(bot);
+
+    // Right click and terminate thread
+    @SuppressWarnings("rawtypes")
+    Matcher matcher = WidgetMatcherFactory.withPartName("Debug");
+    @SuppressWarnings("unchecked")
+    SWTBotView debug = bot.view(matcher);
+    final SWTBotTree tree = debug.bot().tree();
+    SWTBotTreeItem[] items = tree.getAllItems();
+    if (items.length > 0) {
+      SWTBotTreeItem first = items[0];
+      first.contextMenu("Terminate").click();
+
+      // Wait for process to spin down
+      bot.waitUntil(new DefaultCondition() {
+        @Override
+        public String getFailureMessage() {
+          return "Couldn't find terminated process.";
+        }
+
+        @Override
+        public boolean test() throws Exception {
+          SWTBotTreeItem[] items = tree.getAllItems();
+          return items[0].getText().contains("terminated");
+        }
+      });
+    }
+  }
+
+  public static void openJavaPerpsective(SWTBot bot) {
```

Comment:
the `openDebugPerspective` method is more complicated than this. Could not the two methods use the same logic, have one method that accepts the bot and the perspective name and then separate one line methods for invoking it with `Java`/`Debug` (and the bot)?